addon/factory.js

```diff
@@ -4,29 +4,21 @@ import _isFunction from 'lodash/lang/isFunction';
 import _isPlainObject from 'lodash/lang/isPlainObject';
 import _mapValues from 'lodash/object/mapValues';
 
+function buildValue(value, parentAttrs, sequence) {
+  if (_isArray(value)) {
+    return value.map((val) => buildValue(val, value, sequence));
+  } else if (_isPlainObject(value)) {
+    return _mapValues(value, (val) => buildValue(val, value, sequence));
+  } else if (_isFunction(value)) {
+    return value.call(parentAttrs, sequence);
+  } else {
+    return value;
+  }
+}
+
```

Comment:
I'm finding this code difficult to read. In the future I'd like to see

 - `Factory` become an ES6 class, so we can easily add new methods on it
 - fewer named functions divorced from the class